com.unity.multiplayer.mlapi/Runtime/NetworkVariable/Collections/NetworkDictionary.cs

```diff
@@ -403,7 +404,9 @@ public bool IsDirty()
             get => m_Dictionary[key];
             set
             {
-                if (NetworkManager.Singleton.IsServer)
+                EnsureInitialized();
```

Comment:
I suppose we only need these EnsureInitialized inside the IsServer block, yes?

Reply:
But in general I'm a little confused.  This is an instance method, and we have m_NetworkBehaviour being initialized in the xtor (as private and readonly).  Are we thinking somehow it could be nullified after the xtor?  if it could be null  at xtor time, then I would think we should throw an exception then.  Concern is code clarity and the cost of doing these checks on every access.